### DISCUSSION TITLE
[Q&A] 代理服务器地址填了聊天没任何反应

### DISCUSSION THREAD

**Discussion by irwinai** [lobehub/lobehub#1065] [General Question] [answered]
### 💻 Operating System

macOS

### 📦 Environment

Vercel / Zeabur / Sealos

### 🌐 Browser

Chrome

### 🐛 Bug Description

在next-chat中使用的也是一样的代理地址和key，可以使用，这里填了就是不行，接口200，返回内容是空的

### 🚦 Expected Behavior

在next-chat中使用的也是一样的代理地址和key，可以使用，这里填了就是不行，接口200，返回内容是空的

### 📷 Recurrence Steps

在next-chat中使用的也是一样的代理地址和key，可以使用，这里填了就是不行，接口200，返回内容是空的

### 📝 Additional Information

在next-chat中使用的也是一样的代理地址和key，可以使用，这里填了就是不行，接口200，返回内容是空的

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
确认是否 URL 包含了 `/v1`

[a comment by an automated account is not shown]

**Comment by irwinai** [comment]
> 确认是否 URL 包含了 `/v1`

确实。。为什么做这种设计，文档也没有说明。。。

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
> 文档也没有说明。。。

文档里不是提了么？

<img width="1097" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/ddb0a279-32ca-4153-8598-06449fd3fe5e">

> 为什么做这种设计

这个跟模型服务商有很大关系，比如 openai 的默认地址是 `openai.com/v1` 。如果你的代理上是转发了 v1 这个接口，那么你就直接填 proxy.com 就好了，不用填 `/v1` 。 但如果模型服务商是直接转发了 openai.com 域名，那你就要自己加上 `/v1` 这个 url。

----
为什么 ChatNextWeb 不用，而 LobeChat 需要？

ChatNextWeb 的做法是直接写死了请求路径，自己构造了http请求直接往这个url 打。

https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/blob/main/app/constant.ts#L79

它的问题就是如果后续 openai 升级了接口版本到了 V2 ，那这个模式就没法兼容了。

但我们在 LobeChat 里目前使用的是 OpenAI 官方 npm 包。他们的 baseURL 参数里可以指定版本，因此对长远来看的兼容性会更好。
